saleor/graphql/product/mutations/products.py

```diff
@@ -236,6 +259,21 @@ class Meta:
         description = 'Creates a new product.'
         model = models.Product
 
+    @classmethod
+    def clean_default_variant(cls, cleaned_input, product_type, errors):
+        default_variant = cleaned_input.pop('default_variant', None)
+        if default_variant and product_type.has_variants:
+            cls.add_error(
+                errors, 'default_variant', 'Product type has variants')
+        elif default_variant and not product_type.has_variants:
+            cleaned_input['default_variant'] = \
+                ProductVariantCreate.clean_attributes_input(
```

Comment:
Shouldn't we call `ProductVariantCreate. clean_input` here? From reading this code I assume that `clean_attributes_input` would only clean the attributes data, while it should clean the whole variant data.